providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -186,7 +186,14 @@ private <T> ProviderEvaluation<T> resolve(Class<T> type, String key,
             log.debug(message);
             throw new TypeMismatchError(message);
         }
-
+        // if this is an integer and we are trying to resolve a double, convert
+        if (value instanceof Integer && type == Double.class) {
+            value = ((Integer) value).doubleValue();
+        
+        // if this is a doulbe and we are trying to resolve an integer, convert
+        } else if (value instanceof Double && type == Integer.class) {
+            value = ((Double) value).intValue();
+        }
```

Comment:
```suggestion
        if (value instanceof Integer && type == Double.class) {
            // if this is an integer and we are trying to resolve a double, convert
            value = ((Integer) value).doubleValue();
        } else if (value instanceof Double && type == Integer.class) {
            // if this is a double and we are trying to resolve an integer, convert
            value = ((Double) value).intValue();
        }
```

suggestion to improve readablity